resource/gradient-color-scale.js

```diff
@@ -0,0 +1,58 @@
+
+
+function findId(data, idToLookFor) {
+    var categoryArray = data.palettes;
+    for (var i = 0; i < categoryArray.length; i++) {
+        if (categoryArray[i].id === idToLookFor) {
+            return(categoryArray[i].colors);
+        }
+    }
+}
+
+
+function convertBuiltInPaletteToGradient(colorPalette){
+    let sample = findId(dkuColorPaletteContinuous, colorPalette);
+    let myColor = d3.scaleQuantize()
+        .range(sample)
+        .domain([0, 10]);
+    gradient = {};
+    for (let i=0; i <= 9; i++) {
+        gradient[i/10] = myColor(i);
+    }
+    return gradient
+}
+
+/**
+ * Convert the name of the palette to the leaflet heat map gradient object
+ * @param colorPalette: Name of the input colorPalette
+ * @returns {{}}
+ */
+function convertPaletteToGradient(colorPalette){
+    var isFirstLetterUppercase = /^[A-Z]/.test(colorPalette);
+    if (isFirstLetterUppercase) return convertColorBrewerPaletteToGradient(colorPalette)
+    else return convertBuiltInPaletteToGradient(colorPalette)
+}
```

Comment:
Great ternary operator for this kind of things:
```suggestion
   return isFirstLetterUppercase ? convertColorBrewerPaletteToGradient(colorPalette) : convertBuiltInPaletteToGradient(colorPalette)
}
```

Or to be even more concise
```suggestion
    return (isFirstLetterUppercase ? convertColorBrewerPaletteToGradient : convertBuiltInPaletteToGradient)(colorPalette);
}
```

Reply:
Done in 0bed5c7